File: iosApp/iosApp/UI/Screens/Player/Models/MediaTrack.swift
import Foundation

struct MediaTrack: Identifiable, Equatable {
    let id = UUID()
    let label: String
    let isSelected: Bool
    let isAuto: Bool
    var bitrate: Double = 0

    static func == (lhs: MediaTrack, rhs: MediaTrack) -> Bool {
        lhs.label == rhs.label && lhs.isSelected == rhs.isSelected && lhs.isAuto == rhs.isAuto && lhs.bitrate == rhs.bitrate
    }
}

struct AvailableTracks {
    var videoTracks: [MediaTrack] = []
    var audioTracks: [MediaTrack] = []
}

File: androidDemo/src/main/java/com/novage/p2pml/demo/ui/screens/player/models/MediaTrack.kt
package com.novage.p2pml.demo.ui.screens.player.models

import androidx.compose.runtime.Immutable

@Immutable
data class MediaTrack(
    val label: String,
    val isSelected: Boolean,
    val groupIndex: Int,
    val trackIndex: Int,
    val trackType: Int,
    val isAuto: Boolean = false
)

@Immutable
data class AvailableTracks(
    val videoTracks: List<MediaTrack> = emptyList(),
    val audioTracks: List<MediaTrack> = emptyList()
)
